src/Microsoft.OData.Core/MultipartMixed/ODataMultipartMixedBatchWriter.cs

```diff
@@ -0,0 +1,404 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataMultipartMixedBatchWriter.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.MultipartMixed
+{
+    #region Namespaces
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Globalization;
+#if PORTABLELIB
+    using System.Threading.Tasks;
+#endif
+    #endregion Namespaces
+
+    internal sealed class ODataMultipartMixedBatchWriter : ODataBatchWriter
+    {
+        /// <summary>The boundary string for the batch structure itself.</summary>
+        private readonly string batchBoundary;
+
+        /// <summary>
+        /// Gets the writer's output context as the real runtime type.
+        /// </summary>
+        private readonly ODataRawOutputContext rawOutputContext;

```

Comment:
As with the reader; I would rather remove this private property, and just access a protected get InputContext from the base class.  We can have a private get in the derived class that casts, if necessary.

Reply:
Sure we can, it has no functional impact. That will add context base type & make it visible on the base class interface, with additional interaction between derived class and base class.
Making the proposed change and resolving now, please write back in case you have second thought.

---
In reply to: [145856672](https://github.com/OData/odata.net/pull/960#discussion_r145856672) [](ancestors = 145856672)